packit/cli/builds/obs_build.py

```diff
@@ -0,0 +1,309 @@
+                )
+                added = True
+
+        if not added:
+            repos.append(Repository(name=name, path=path, arch=[arch]))
+
+    root = ET.Element("project")
+    root.attrib["name"] = prj_name
+
+    (title_elem := ET.Element("title")).text = "Packit project"
+    (descr_elem := ET.Element("description")).text = description or ""
+    (person_elem := ET.Element("person")).attrib["userid"] = owner
+    person_elem.attrib["role"] = "maintainer"
+
+    for elem in (title_elem, descr_elem, person_elem):
+        root.append(elem)
+
+    for repo in repos:
+        (repo_elem := ET.Element("repository")).attrib["name"] = repo.name
+
+        for path_entry in repo.path:
+            (path_elem := ET.Element("path")).attrib["project"] = path_entry.project
+            path_elem.attrib["repository"] = path_entry.repository
+            repo_elem.append(path_elem)
+
+        for arch in repo.arch:
+            (arch_elem := ET.Element("arch")).text = arch
+            repo_elem.append(arch_elem)
+
+        root.append(repo_elem)
+
+    return root
+
+
+def create_package(prj_name: str, pkg_name: str) -> None:
+    """Creates the package with the name ``pkg_name`` in the project
+    ``prj_name`` on OBS. No sources are uploaded in the process.
+
+    """
+    (root := ET.Element("package")).attrib["name"] = pkg_name
+    root.attrib["project"] = prj_name
+
+    (title := ET.Element("title")).text = f"The {pkg_name} package"
+    descr = ET.Element("description")
+
+    root.append(title)
+    root.append(descr)
+
+    url = core.makeurl(_API_URL, ["source", prj_name, pkg_name, "_meta"])
+    mf = core.metafile(url, ET.tostring(root))
+    mf.sync()
+
+
+@click.command("in-obs", context_settings=get_context_settings())
+@click.option(
+    "--owner",
+    help="OBS user, owner of the project. (defaults to the username from the oscrc)",
+)
+@click.option(
+    "--project",
+    help="Project name to build in. It will be created if does not exist."
+    " It defaults to home:$owner:packit:$pkg",
+)
+@click.option(
+    "--targets",
+    help="Comma separated list of chroots to build in. (defaults to 'fedora-rawhide-x86_64')",
+    default="fedora-rawhide-x86_64",
+)
+@click.option(
+    "--description",
+    help="Description of the project to build in.",
+    default=None,
+)
+@click.option(
+    "--upstream-ref",
+    default=None,
+    help="Git ref of the last upstream commit in the current branch "
+    "from which packit should generate patches "
+    "(this option implies the repository is source-git).",
+)
+@click.option("--wait/--no-wait", default=True, help="Wait for the build to finish")
+@click.option(
+    PACKAGE_SHORT_OPTION,
+    PACKAGE_LONG_OPTION,
+    multiple=True,
+    help=PACKAGE_OPTION_HELP.format(action="build"),
+)
+@click.argument("path_or_url", type=LocalProjectParameter(), default=os.path.curdir)
+@pass_config
+@cover_packit_exception
+@iterate_packages
+def obs(
```

Comment:
also, I would probably move most of this function to be a method of the PackitAPI - this way it can be also later potentially reused by Packit service (you can check e.g. other commands in `cli/builds`)